docs/src/main/asciidoc/elytron-properties-guide.adoc

```diff
@@ -0,0 +1,168 @@
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Elytron Properties File Config
+
+include::./attributes.adoc[]
+
+Quarkus provides support for properties file based authentication that is intended for
+development and testing purposes. It is not recommended that this be used in production as at present only
+plaintext passwords are supported.
+
+Add the following to your `pom.xml`:
+
+[source,xml]
+----
+<dependencies>
+    <!-- Elytron Security extension -->
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-elytron-security-properties-file</artifactId>
+    </dependency>
+</dependencies>
+----
+
+== Configuration
+
+The elytron-security-properties-file extension currently supports two different realms for the storage of authentication
+and authorization information. Both support storage of this information in properties type files. The next two sections detail the specific configuration properties.
+
+=== Property Files Realm Configuration
+
+The property files realm supports mapping of users to password and users to roles with a combination of properties files. To enable and configure it, the following configuration properties are used:
+
+|===
+|Property Name|Default|Description
+|quarkus.users.file.enabled|false|Determine whether security via the file realm is enabled
+|quarkus.users.file.auth-mechanism|BASIC|Name of authentication mechanism to use
+|quarkus.users.file.realm-name|Quarkus|Name to assign the security realm
+|quarkus.users.file.users|users.properties|Classpath resource name of properties file containing user to password mappings; see <<Users.properties>>
+|quarkus.users.file.roles|roles.properties|Classpath resource name of properties file containing user to role mappings; see <<Roles.properties>>
+|===
+
+.example application.properties file section for property files realm
+[source,properties]
+----
+quarkus.users.file.enabled=true
+quarkus.users.file.users=test-users.properties
+quarkus.users.file.roles=test-roles.properties
+quarkus.users.file.auth-mechanism=BASIC
+quarkus.users.file.realm-name=MyRealm
+quarkus.users.file.plain-test=true
+----
+
+==== Users.properties
+
+The `quarkus.users.file.users` configuration property specifies a classpath resource which is a properties file with a user to password mapping, one per line. The following <<test-users-example>> illustrates the format:
+
+[#test-users-example]
+.example test-users.properties file
+[source,properties]
+----
+scott=jb0ss <1>
+jdoe=p4ssw0rd <2>
+stuart=test
+noadmin=n0Adm1n
+----
+<1> User `scott` has password defined as `jb0ss`
+<2> User `jdoe` has password defined as `p4ssw0rd`
+
+This file has the usernames and passwords stored in plain text, which is not recommended. If plain-text is set to false
+(or omitted) in the config then passwords must be stored in the form `MD5 ( username : realm : password )`. This can
```

Comment:
Is there any other hash available? `md5` is really not considered secured these days.

Reply:
LegacyProperyRealm in Elytron only supports this, and it's not supposed to be used in production anyway. The other advantage is that lots of systems have md5 available so you can generate the password by with echo -n user:realm:password | md5.

We could change it but it would be a bit of work as it would mean forking the Elytron implementation, and we would also need to come up with some kind of tool to enable the users to generate the passwords ( I want to do this eventually anyway, but not as an ad-hoc thing).

Reply:
Yeah, probably not worth it. It's just that having `md5` in a security related documentation could look like a joke.